eland/tests/dataframe/test_metrics_pytest.py

```diff
@@ -424,3 +425,50 @@ def test_aggs_count(self):
         ed_count = ed_flights.agg(["count"])
 
         assert_frame_equal(pd_count, ed_count)
+
+    @pytest.mark.parametrize("quantiles", [[0.2, 0.5], [0, 1]])
+    @pytest.mark.parametrize("numeric_only", [False, None])
+    def test_flights_quantile(self, quantiles, numeric_only):
+        pd_flights = self.pd_flights()
+        ed_flights = self.ed_flights()
+
+        pd_quantile = pd_flights.filter(self.filter_data[0:3]).quantile(
+            q=quantiles, numeric_only=numeric_only
+        )
+        ed_quantile = ed_flights.filter(self.filter_data[0:3]).quantile(
+            q=quantiles, numeric_only=numeric_only
+        )
+
+        assert_frame_equal(pd_quantile, ed_quantile, check_exact=False, rtol=2)
+
+        pd_quantile = pd_flights[["timestamp"]].quantile(
+            q=quantiles, numeric_only=numeric_only
+        )
+        ed_quantile = ed_flights[["timestamp"]].quantile(
+            q=quantiles, numeric_only=numeric_only
+        )
+
+        pd_timestamp = pd.to_numeric(pd_quantile.squeeze(), downcast="float")
+        ed_timestamp = pd.to_numeric(ed_quantile.squeeze(), downcast="float")
+
+        assert_series_equal(pd_timestamp, ed_timestamp, check_exact=False, rtol=2)
+
+    @pytest.mark.parametrize("quantiles", [5, [2, 1], -1.5, [1.2, 0.2]])
+    def test_flights_quantile_error(self, quantiles):
+        ed_flights = self.ed_flights().filter(self.filter_data)
+
+        match = f"quantile should be in range of 0 and 1, given {quantiles[0] if isinstance(quantiles, list) else quantiles}"
+        with pytest.raises(ValueError, match=match):
+            ed_flights[["timestamp"]].quantile(q=quantiles)
+
+    @pytest.mark.parametrize("numeric_only", [True, False, None])
+    def test_flights_agg_quantile(self, numeric_only):
+        pd_flights = self.pd_flights().filter(self.filter_data[0:3])
```

Comment:
Instead of using a slice here, prefer to be explicit in the fields being used
`.filter(["AvgTicketPrice", "Cancelled", "dayOfWeek"])`. That way if `self.filter_data` changes we don't get mysterious failures in this test.